### ISSUE TITLE
Add official docker image

### ISSUE THREAD

**Issue by yajo** [fish-shell/fish-shell#3440]
I see you already have a Dockerfile in the repo, so this issue is just for asking you to please simply create an automated build in the Docker Hub, and reference it here in the README.

Thanks!

**Comment by edouard-lopez** [comment]
You can find docker images under [ohmyfish/fish/dockerfile/](https://hub.docker.com/r/ohmyfish/fish/dockerfile/) (up to `3.0.0`). 

I'm submitted a PR to @sagebind https://github.com/oh-my-fish/dockerfiles/pull/9 to simplify the process of building images for new versions, but I'm still waiting for a feedback :S (feel free to weight in).

**Comment by faho** [comment]
Okay, I don't think I understand what the dockerfile is supposed to do.

Isn't docker supposed to be for, you know, full containers? Having just a shell in a container seems less than useful.

Also, as an important caveat:

>You can find docker images under ohmyfish/fish/dockerfile/ (up to 3.0.0).

This means it's almost two years out of date, and missing some important bug fixes (it doesn't even go up to 3.0.2, not to mention 3.1.2).

**Comment by edouard-lopez** [comment]
@faho,

I'm using [docker image to run test](https://github.com/rafaelrinaldi/pure/blob/master/.travis.yml) in a controlled environment for the [_pure_ project](https://github.com/rafaelrinaldi/pure/).

Sadly, `ohmyfish` images are pretty old, and the maintainer doesn't seem active anymore on this repo, and my PR has been pending for 1½ years. 

However, there is a **[repo with up-to-date images!](https://github.com/andreiborisov/docker-fish)** :tada:  (thanks to @andreiborisov).

**Comment by faho** [comment]
I don't see us volunteering to take on yet another repo, and the current situation works fine in practice. Closing!

**Comment by edouard-lopez** [comment]
Hello,
> TL;DR: I maintain [docker images for various versions of Fish](https://github.com/pure-fish/docker-fish/) using Alpine.

I'm the maintainer of [pure prompt](https://github.com/pure-fish/pure/) for Fish and love to test my projects. Until a few months back, I was using [andreiborisov' docker images](https://github.com/andreiborisov/docker-fish) to run `pure` tests in isolated context (thanks so much :). However, new Fish versions were missing, and the build process was too complex to update for the time I had and what I needed:

1. a small container with Fish in major versions
2. easy to update (kids take time :P)
3. provide `fisher` and `fishtape` for the tests

So, I forked the project and went for a simpler approach and **use the Alpine version that provides the Fish version I want** (e.g. Alpine `3.11` provides Fish `3.0.2-r5`). This way I'm leveraging the work done by Alpine contributors and get a workflow that is simple to maintain. ✌️‍



Hope others will find it useful!
